### PR TITLE
Game data model

### PR DESCRIPTION
**Pull request by Robsteranium**
Introduces game logic into `bild-ord.domain.words` (we might want to move it to e.g. `domain.game` or something). See tests for demonstration. This allows us to keep e.g. the correctness-check separate from the correctness-representation.

The views needed quite a bit of modification - e.g. using `map` over a collection of questions rather than creating subscriptions to each word with `for`. I don't know what the relative merits are of the two approaches - this was just what came into my head!

This PR could be merged as is (i.e. no loss of functionality) or be reviewed and dissected to figure out a better way forward...

### CODE REVIEW

File: test/bild_ord/domain/words_test.clj [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+             (responses game))))
+    (testing "is not a success"
+      (is (not (success? game))))))
+
+(deftest correctness-test
+  (testing "A guess may be correct"
+    (is (true? (correct? {"lura" "lura"}))))
+  (testing "A guess may be incorrect"
+    (is (false? (correct? {"lura" "sila"}))))
+  (testing "A game is a collection of correctness"
```
Comment (jstaffans) [inline test/bild_ord/domain/words_test.clj:38]: "collection of correctness" will be my next band name!

**Comment by jstaffans** [comment]
Good stuff! I like keeping the state (questions and answers) really separate from the view logic.

One suggestion I have is to use the plumatic/schema library to ensure that the game state stays valid between transitions. I find it a lot easier to understand what's going on when there is at least some schema for the most important bits of data. What do you think about that?

**Comment by Robsteranium** [comment]
Yeah that's a great suggestion. I wonder if it might help with detecting wiring mistakes in re-frame too...

**Comment by jstaffans** [comment]
The re-frame state atom could also be subjected to a schema. Definitely helpful for understanding those parts as well. Was that what you referred to as "wiring"?

**Comment by jstaffans** [comment]
I can totally see us bumping to the latest Clojure and ClojureScript versions and using clojure.spec!

**Comment by jstaffans** [comment]
Anyway, go ahead and merge this PR -- we can get back to the state validation and clojure.spec later on. You can also delete all the stuff that you commented out with `#_` reader tags.

**Comment by Robsteranium** [comment]
By "wiring" I was referring to the fact that you can, as I have (twice), create a subscription to something that doesn't exist in the db atom. For example e.g. `(reaction (:unavailable-key @db))` will return `nil`. This means you either get seemingly random NPEs or, worse still, no warning that anything is wrong at all (except of course that the app won't work!).

I imagine that some sort of type checking might catch those sort of bugs.

Yes let's give clojure.spec a try!

I can merge the PR in my repo but I don't have permissions to this one. I'll prepare another commit to remove the commented-out code.

**Comment by jstaffans** [comment]
I did invite you as a collaborator, it seems you didn't respond to the invitation yet. Did it go missing?

**Comment by Robsteranium** [comment]
Ah great, thanks. Sorry I'd not seen the notification.

**Pull request merged**